File: backend/src/main/java/es/udc/paproject/backend/model/services/StretchingService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Stretching;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;
import es.udc.paproject.backend.model.exceptions.UsedStretchingException;

public interface StretchingService {
    
    Stretching addStretching(String stretchingName, String description, String stretchingType) throws InstanceNotFoundException;

    void addStretchingToPlayer(Long playerId, Long stretchingId) throws InstanceNotFoundException;

    void addStretchingToTraining(Long trainingId, Long stretchingId) throws InstanceNotFoundException;

    void addStretchingToGame(Long gameiId, Long stretchingId) throws InstanceNotFoundException;

    Stretching findStretchingById(Long stretchingId) throws InstanceNotFoundException;

    List<Stretching> findAllStretchings() throws InstanceNotFoundException;

    Block<Stretching> findAllStretchings(int page, int size) throws InstanceNotFoundException;

    Block<Stretching> findStretchingsByType(String stretchingType, int page, int size) throws InstanceNotFoundException;

    List<Stretching> findStretchingsByType(String stretchingType) throws InstanceNotFoundException;

    List<Stretching> findStretchingsByPlayerId(Long playerId) throws InstanceNotFoundException;

    List<Stretching> findStretchingsByTrainingId(Long trainingId) throws InstanceNotFoundException;

    List<Stretching> findStretchingsByGameId(Long gameId) throws InstanceNotFoundException;

    void removeStretching(Long stretchingId) throws InstanceNotFoundException, UsedStretchingException;

    void removeStretchingToPlayer(Long stretchingId, Long playerId) throws InstanceNotFoundException;

    void removeStretchingToTraining(Long stretchingId, Long trainingId) throws InstanceNotFoundException;

    void removeStretchingToGame(Long stretchingId, Long gameId) throws InstanceNotFoundException;

    Stretching updatStretching(Long stretchingId, String stretchingName, String description, String stretchingType) throws InstanceNotFoundException;
}

File: frontend/src/backend/stretchingService.js
import {config, appFetch} from './appFetch';

export const findStretchingById = (stretchingId, onSuccess, onErrors) => 
    appFetch(`/stretchings/${stretchingId}`, config('GET'), onSuccess, onErrors);

export const findAllStretchings = (onSuccess, onErrors) => 
    appFetch('/stretchings', config('GET'), onSuccess, onErrors);

export const findAllStretchingsPage = ({page}, onSuccess, onErrors) => 
    appFetch(`/stretchings/page?page=${page}`, config('GET'), onSuccess, onErrors);

export const findStretchingsByTypePage = ({stretchingType, page}, onSuccess, onErrors) =>
    appFetch(`/stretchings/${stretchingType}/stretchingType/page?page=${page}`, config('GET'), onSuccess, onErrors);

export const findStretchingsByType = (stretchingType, onSuccess, onErrors) =>
    appFetch(`/stretchings/${stretchingType}/stretchingType`, config('GET'), onSuccess, onErrors);

export const findStretchingsByPlayerId = (playerId, onSuccess, onErrors) => 
    appFetch(`/stretchings/${playerId}/player`, config('GET'), onSuccess, onErrors);

export const findStretchingsByTrainingId = (trainingId, onSuccess, onErrors) => 
    appFetch(`/stretchings/${trainingId}/training`, config('GET'), onSuccess, onErrors);

export const findStretchingsByGameId = (gameId, onSuccess, onErrors) => 
    appFetch(`/stretchings/${gameId}/game`, config('GET'), onSuccess, onErrors);

export const addStretching = (stretchingName, description, stretchingType, onSuccess, onErrors) => 
    appFetch(`/stretchings?stretchingName=${stretchingName}&description=${description}&stretchingType=${stretchingType}`, config('POST'), onSuccess, onErrors);

export const addStretchingToPlayer = (playerId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${playerId}/addStretchingToPlayer?stretchingId=${stretchingId}`, config('POST'), onSuccess, onErrors);

export const addStretchingToTraining = (trainingId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${trainingId}/addStretchingToTraining?stretchingId=${stretchingId}`, config('POST'), onSuccess, onErrors);

export const addStretchingToGame = (gameId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${gameId}/addStretchingToGame?stretchingId=${stretchingId}`, config('POST'), onSuccess, onErrors);

export const updatStretching = (stretchingId, stretchingName, description, stretchingType, onSuccess, onErrors) => 
    appFetch(`/stretchings/${stretchingId}?stretchingName=${stretchingName}&description=${description}&stretchingType=${stretchingType}`, config('PUT'), onSuccess, onErrors);

export const removeStretching = (stretchingId, onSuccess, onErrors) => 
    appFetch(`/stretchings/${stretchingId}`, config('DELETE'), onSuccess, onErrors);

export const removeStretchingToPlayer = (playerId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${playerId}/player?stretchingId=${stretchingId}`, config('DELETE'), onSuccess, onErrors);

export const removeStretchingToTraining = (trainingId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${trainingId}/training?stretchingId=${stretchingId}`, config('DELETE'), onSuccess, onErrors);

export const removeStretchingToGame = (gameId, stretchingId, onSuccess, onErrors) =>
    appFetch(`/stretchings/${gameId}/game?stretchingId=${stretchingId}`, config('DELETE'), onSuccess, onErrors);